core/trino-main/src/main/java/io/trino/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -500,6 +501,50 @@ public PlanWithProperties visitWindow(WindowNode node, HashComputationSet parent
                     child.getHashSymbols());
         }
 
+        @Override
+        public PlanWithProperties visitTableFunctionProcessor(TableFunctionProcessorNode node, HashComputationSet parentPreference)
+        {
+            return visitPlan(node, parentPreference);
+
+            // TODO The below code adds a hash symbol. The optimization fails because the new symbol is no in the node's output
+            //  (it is neither a proper output nor a pass-through symbol). This could be fixed by exposing the symbol on output,
+            //  but neither the proper outputs nor the pass-through is the right place to do that.
+            /*if (node.getSource().isEmpty()) {
```

Comment:
What's the point of keeping this code around?

Reply:
removed, hash generation should be added as a follow-up.